src/span.h

```diff
@@ -236,11 +237,35 @@ T& SpanPopBack(Span<T>& span)
     return back;
 }
 
+// From C++20 as_bytes and as_writeable_bytes
+template <typename T>
+Span<const std::byte> AsBytes(Span<T> s) noexcept
+{
+    return {reinterpret_cast<const std::byte*>(s.data()), s.size_bytes()};
+}
+template <typename T>
+Span<std::byte> AsWriteableBytes(Span<T> s) noexcept
+{
+    return {reinterpret_cast<std::byte*>(s.data()), s.size_bytes()};
+}
+
+template <typename V>
+Span<const std::byte> MakeByteSpan(V&& v) noexcept
+{
+    return AsBytes(MakeSpan(std::forward<V>(v)));
+}
+template <typename V>
+Span<std::byte> MakeWriteableByteSpan(V&& v) noexcept
```

Comment:
I am getting a compilation error while trying to write a unit test for this function.

Test:
https://github.com/klementtan/bitcoin/blob/7f8a29b29039710955d81d820fad15d0156e95f3/src/test/hash_tests.cpp#L180

Compilation Error:
```
./span.h:249:13: error: reinterpret_cast from 'const char *' to 'std::byte *' casts away qualifiers
    return {reinterpret_cast<std::byte*>(s.data()), s.size_bytes()};
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./span.h:260:12: note: in instantiation of function template specialization 'AsWriteableBytes<const char>' requested here
    return AsWriteableBytes(MakeSpan(std::forward<V>(v)));
           ^
test/hash_tests.cpp:180:46: note: in instantiation of function template specialization 'MakeWriteableByteSpan<char const (&)[7]>' requested here
  Span<std::byte> writeable_byte_span_char = MakeWriteableByteSpan("foobar");
```